includes/reports/reports-functions.php

```diff
@@ -0,0 +1,453 @@
+	if ( ! isset( $tile['args'] ) ) {
+		return;
+	}
+
+	if ( empty( $tile['args']['data'] ) ) {
+		echo '<span class="tile-no-data tile-value">' . __( 'No data for the current date range.', 'easy-digital-downloads' ) . '</span>';
+	} else {
+		switch( $tile['args']['display_args']['type'] ) {
+			case 'number':
+				echo '<span class="tile-number tile-value">' . edd_format_amount( $tile['args']['data'] ) . '</span>';
+				break;
+
+			case 'split-number':
+				printf( '<span class="tile-amount tile-value">%1$d / %2$d</span>',
+					edd_format_amount( $tile['args']['data']['first_value'] ),
+					edd_format_amount( $tile['args']['data']['second_value'] )
+				);
+				break;
+
+			case 'split-amount':
+				printf( '<span class="tile-amount tile-value">%1$d / %2$d</span>',
+					edd_currency_filter( edd_format_amount( $tile['args']['data']['first_value'] ) ),
+					edd_currency_filter( edd_format_amount( $tile['args']['data']['second_value'] ) )
+				);
+				break;
+
+			case 'amount':
+				echo '<span class="tile-amount tile-value">' . edd_currency_filter( edd_format_amount( $tile['args']['data'] ) ) . '</span>';
+				break;
+
+			case 'url':
+				echo '<span class="tile-url tile-value">' . esc_url( $tile['args']['data'] ) . '</span>';
+				break;
+
+			default:
+				echo '<span class="tile-value">' . esc_html( $tile['args']['data'] ) . '</span>';
+				break;
+		}
+	}
+
+	if ( ! empty( $tile['args']['display_args']['comparison_label'] ) ) {
+		echo '<span class="tile-compare">' . $tile['args']['display_args']['comparison_label'] . '</span>';
+	}
+}
+
+/**
+ * Handles default display of all tile endpoints registered against a report.
+ *
+ * @since 3.0
+ *
+ * @param \EDD\Reports\Data\Report $report Report object.
+ */
+function default_display_tiles_group( $report ) {
+	if ( $report->has_endpoints( 'tiles' ) ) : ?>
+
+		<div id="edd-reports-tiles-wrap">
+			<h3><?php _e( 'Quick Stats', 'easy-digital-downloads' ); ?></h3>
+
+			<div id="dashboard-widgets" class="metabox-holder">
+
+				<div class="postbox-container">
+					<?php do_meta_boxes( 'download_page_edd-reports', 'primary', $report ); ?>
+				</div>
+
+				<div class="postbox-container">
+					<?php do_meta_boxes( 'download_page_edd-reports', 'secondary', $report ); ?>
+				</div>
+
+				<div class="postbox-container">
+					<?php do_meta_boxes( 'download_page_edd-reports', 'tertiary', $report ); ?>
+				</div>
+
+			</div>
+		</div>
+	<?php endif; // Has endpoints.
+}
+
+/**
+ * Handles default display of all table endpoints registered against a report.
+ *
+ * @since 3.0
+ *
+ * @param \EDD\Reports\Data\Report $report Report object.
+ */
+function default_display_tables_group( $report ) {
+	if ( $report->has_endpoints( 'tables' ) ) :
+		$tables = $report->get_endpoints( 'tables' );
+		?>
+		<div id="edd-reports-tables-wrap">
+
+			<?php foreach ( $tables as $endpoint_id => $table ) : ?>
+				<h3><?php echo esc_html( $table->get_label() ); ?></h3>
```

Comment:
You probably know this already, but `$table` is an array here so this fatals.

Reply:
`get_endpoints()` returns an array of keyed objects. If `$table` is an array here, then something else is wrong. I think I may have addressed the problem in 8f19622607156b2b83e5a67d3931585856418942.

Reply:
Looks like that commit fixed it.